src/mscorlib/model.xml

```diff
@@ -683,6 +683,9 @@
       <Member Name="GetHashCode(T)" />
       <Member MemberType="Property" Name="Default" />
     </Type>
+    <Type Name="System.Collections.Generic.EnumEqualityComparer&lt;T&gt;">
```

Comment:
EnumEqualityComparer's GetObjectData uses SerializationInfo.SetType to change what type it'll be deserialized as if the enum's base isn't Int32:
https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Collections/Generic/EqualityComparer.cs#L393
As a result, none of those deserialization ctors on those derived types are used; they're stripped out by the rewriter, and that's ok.  We could delete them from the source I expect as well (though I expect there are a bunch of things we could remove rather than relying on the rewriter to do it).